src/components/modals/RegistrationModal.tsx

```diff
@@ -1,20 +1,76 @@
 import React from 'react';
 import { CardActions } from '@mui/material';
 import { useTranslation } from 'react-i18next';
+import { Controller } from 'react-hook-form';
 import Input from '../common/input/Input';
 import Button from '../common/button/Button';
 import Form from '../layout/form/Form';
+import { useFormWithSchema } from '../../hooks/useYupValidationResolver';
+import { registrationSchema } from '../../shemas';
 
 const RegistrationModal = () => {
   const { t } = useTranslation('common');
+  const { handleSubmit, control } = useFormWithSchema(registrationSchema);
 
   return (
-    <Form>
-      <Input label="Email" />
-      <Input label={t('password')} />
-      <Input label={t('repeatPassword')} type="password" />
+    <Form onSubmit={handleSubmit((data) => console.log(data, 'data'))}>
+      <Controller
```

Comment:
I'd suggest creating an array with all needed information for each Controller component and using map() method to render them. By doing this, you will remove repetitive jsx.